packages/workmanager/tizen/src/workmanager_tizen_plugin.cc

```diff
@@ -0,0 +1,481 @@
+
+const char *kEventName = "pass_taskinfo_event";
+
+const char *kInvalidArg = "Invalid argument";
+const char *kOperationFailed = "Operation failed/cancelled";
+
+bool CheckAppIsRunning(const char *app_id) {
+  app_context_h context;
+  int ret = app_manager_get_app_context(app_id, &context);
+  if (ret == APP_MANAGER_ERROR_NO_SUCH_APP) {
+    return false;
+  }
+
+  if (ret != APP_MANAGER_ERROR_NONE) {
+    LOG_ERROR("%s", get_error_message(ret));
+    return false;
+  }
+
+  app_state_e state;
+  app_context_get_app_state(context, &state);
+
+  switch (state) {
+    case APP_STATE_FOREGROUND:
+    case APP_STATE_BACKGROUND:
+    case APP_STATE_SERVICE:
+      return true;
+  }
+  return false;
+}
+
+void SendLaunchRequest(const char *app_id) {
+  app_control_h control;
+  int ret = app_control_create(&control);
+  if (ret != APP_CONTROL_ERROR_NONE) {
+    LOG_ERROR("%s", get_error_message(ret));
+  }
+
+  ret = app_control_set_app_id(control, app_id);
+  if (ret != APP_CONTROL_ERROR_NONE) {
+    LOG_ERROR("%s", get_error_message(ret));
+  }
+
+  ret = app_control_send_launch_request(control, NULL, NULL);
+  if (ret != APP_CONTROL_ERROR_NONE) {
+    LOG_ERROR("%s", get_error_message(ret));
+  }
+
+  ret = app_control_destroy(control);
+  if (ret != APP_CONTROL_ERROR_NONE) {
+    LOG_ERROR("%s", get_error_message(ret));
+  }
+}
+
+bool CheckIsServiceApp() {
+  char *app_id;
+  int ret = app_manager_get_app_id(getpid(), &app_id);
+  if (ret != APP_MANAGER_ERROR_NONE) {
+    LOG_ERROR("Failed to get app id: %s", get_error_message(ret));
+    return false;
+  }
+
+  app_info_h app_info;
+  app_info_app_component_type_e app_type;
+  app_info_create(app_id, &app_info);
+  app_info_get_app_component_type(app_info, &app_type);
+  free(app_id);
+
+  return app_type == APP_INFO_APP_COMPONENT_TYPE_SERVICE_APP;
+}
+
+class WorkmanagerTizenPlugin : public flutter::Plugin {
+ public:
+  static void RegisterWithRegistrar(flutter::PluginRegistrar *registrar) {
+    auto plugin = std::make_unique<WorkmanagerTizenPlugin>();
+
+    is_service_app_ = CheckIsServiceApp();
+
+    if (is_service_app_) {
+      background_channel_ = std::make_unique<FlMethodChannel>(
+          registrar->messenger(), kBackgroundChannelName,
+          &flutter::StandardMethodCodec::GetInstance());
+
+      background_channel_.value()->SetMethodCallHandler(
+          [plugin_pointer = plugin.get()](const auto &call, auto result) {
+            plugin_pointer->HandleBackground(call, std::move(result));
+          });
+
+      event_handler_h handler;
+      std::string service_app_id = GetAppId().value();
+      std::string event_id =
+          "event." + service_app_id.substr(0, service_app_id.size() - 8) + "." +
+          kEventName;
+      event_add_event_handler(event_id.c_str(), TaskInfoCallback, nullptr,
+                              &handler);
+
+      auto &scheduler = JobScheduler::instance();
+      auto job_names = scheduler.GetAllJobs();
```

Comment:
It is not recommended to use auto unless the return type is explicit in the code line. Same for other lines.
Please refer to this https://google.github.io/styleguide/cppguide.html#Type_deduction

Reply:
fixed it!